hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py

```diff
@@ -84,6 +84,7 @@ def check_employee_attendance_state(self):
     def create(self, vals):
         res = super(HrTimesheetSheet, self).create(vals)
         attendances = self.env['hr.attendance'].search([
+            ('employee_id', '=', vals['employee_id']),
```

Comment:
You should get the employee_id from res:

```suggestion
            ('employee_id', '=', res.employee_id.id),
```

and I think this is the reason why the tests are failing.

Reply:
Thanks for this - i've changed this now